### PR TITLE
feat(apigatewayv2): Create HTTP APIs from an OpenAPI specification

### LINKED ISSUE #19190: (module name): short issue description
### Description

Adding OpenAPI support for HTTP APIs (API Gateway V2).

Reference: https://docs.aws.amazon.com/apigateway/latest/developerguide/http-api-open-api.html

### Use Case

It would be great to develop both REST and HTTP API using OpenAPI files (one file per API gateway, of course).

### Proposed Solution

Import the OpenAPI specification per the reference above.

### Other information

_No response_

### Acknowledge

- [X] I may be able to implement this feature request
- [ ] This feature might incur a breaking change

### PR DESCRIPTION
**Pull request by miguel-a-calles-mba**
Adding OpenAPI support for HTTP APIs (API Gateway V2).

Closes #19190
Created a v2 pr at #20815

----

### All Submissions:

* [x] Have you followed the guidelines in our [Contributing guide?](https://github.com/aws/aws-cdk/blob/master/CONTRIBUTING.md)

### Adding new Unconventional Dependencies:

* [ ] This PR adds new unconventional dependencies following the process described [here](https://github.com/aws/aws-cdk/blob/master/CONTRIBUTING.md/#adding-new-unconventional-dependencies)

### New Features

* [ ] Have you added the new feature to an [integration test](https://github.com/aws/aws-cdk/blob/master/INTEGRATION_TESTS.md)?
	* [ ] Did you use `cdk-integ` to deploy the infrastructure and generate the snapshot (i.e. `cdk-integ` without `--dry-run`)?

*By submitting this pull request, I confirm that my contribution is made under the terms of the Apache-2.0 license*

### CODE REVIEW

**Comment by miguel-a-calles-mba** [comment]
@otaviomacedo, the unit tests are failing in the aws-cdk module and not the aws-apigatewayv2 module.

**Comment by miguel-a-calles-mba** [comment]
@otaviomacedo, do you have time to review and provide feedback? Thank you.

**Comment by miguel-a-calles-mba** [comment]
@otaviomacedo, do you have time to review and provide feedback? Thank you.

**Comment by ncaq** [comment]
I have been searching the documentation for this.
I hope it gets merged soon.

File: packages/@aws-cdk/aws-apigatewayv2/lib/http/api-definition.ts [hunk 1]
```diff
@@ -0,0 +1,189 @@
...
+import { Construct } from '@aws-cdk/core';
+
+/**
+ * Represents an OpenAPI definition asset.
+ */
+export abstract class ApiDefinition {
+  /**
+   * Creates an API definition from a specification file in an S3 bucket
+   */
+  public static fromBucket(bucket: s3.IBucket, key: string, objectVersion?: string): S3ApiDefinition {
```
Comment (otaviomacedo) [inline packages/@aws-cdk/aws-apigatewayv2/lib/http/api-definition.ts:23]: A more accurate name would be `fromS3Object`.

**Review by otaviomacedo — COMMENTED** [review]
Looking good. Can you please add documentation about this feature?

**Comment by miguel-a-calles-mba** [comment]
> Looking good. Can you please add documentation about this feature?

Where do I add the documentation?

**Comment by otaviomacedo** [comment]
> Where do I add the documentation?

In the README.md file.

**Comment by TheRealAmazonKendra** [comment]
This PR has been in the CHANGES REQUESTED state for 3 weeks, and looks abandoned. To keep this PR from being closed, please continue work on it. If not, it will automatically be closed in a week.

**Comment by miguel-a-calles-mba** [comment]
@otaviomacedo, does it make sense to abandon this PR and use #20815 instead? This PR was originally created for aws cdk v1. The other PR was built for aws cdk v2. Please advise.

**Comment by TheRealAmazonKendra** [comment]
Closing this in favor of #20815.

**Pull request closed without merging**